pkg/backend/telemetryapi/harvester.go

```diff
@@ -93,11 +100,32 @@ func NewHarvester(options ...func(*Config)) (*Harvester, error) {
 		"version":                version,
 	})
 
+	defer h.Start()
+
+	return h, nil
+}
+
+func (h *Harvester) Start() {
+
 	if 0 != h.config.HarvestPeriod {
 		go harvestRoutine(h)
 	}
 
-	return h, nil
+	for i := 0; i < h.config.MaxConns; i++ {
+		go func(workerNo int) {
+			wlog := logger.WithField("WorkerNo.", workerNo)
+			wlog.Debug("Starting worker.")
+			for {
+				select {
+				case <-h.config.Context.Done():
+					wlog.Debug("Shutting down worker.")
+					return
+				case req := <-h.requestsQueue:
+					harvestRequest(req, &h.config)
+				}
+			}
+		}(i + 1)
```

Comment:
You could also init `i` to 1

Reply:
Valid point 👍 I think a nerve twitches inside of me when a `for` loop doesn't start at 0 😅 Will update this.